### PR TITLE
Ui: Integrate contact into contact detail panel

### PR DESCRIPTION
**Pull request by Tex-Tang**

### CODE REVIEW

**Review by wrewsama — APPROVED** [review]
lgtm

File: src/main/java/seedu/address/ui/ContactBox.java [hunk 1]
```diff
@@ -17,18 +23,33 @@ public class ContactBox extends UiPart<Region> {
...
+        Alert a = new Alert(Alert.AlertType.ERROR);
+        contactLabel.setOnAction(e -> {
+            if (Desktop.isDesktopSupported()) {
+                try {
+                    Desktop.getDesktop().browse(new URI(contact.getLink()));
+                } catch (IOException ex) {
+                    a.setContentText("Sorry, browser couldn't be opened.");
+                    a.show();
+                } catch (URISyntaxException ex) {
+                    a.setContentText("Sorry, url is invalid");
```
Comment (arnav-ag) [inline src/main/java/seedu/address/ui/ContactBox.java:49]: Should we change this to maybe something more appropriate? As this is an internal error moreso than a user error (they probably shouldn't know if this happens)

File: src/main/java/seedu/address/ui/ContactBox.java [hunk 2]
```diff
@@ -17,18 +23,33 @@ public class ContactBox extends UiPart<Region> {
...
-        contactLabel.setText(name);
+        contactLabel.setText(contact.toString());
+
+        Alert a = new Alert(Alert.AlertType.ERROR);
+        contactLabel.setOnAction(e -> {
+            if (Desktop.isDesktopSupported()) {
+                try {
+                    Desktop.getDesktop().browse(new URI(contact.getLink()));
+                } catch (IOException ex) {
+                    a.setContentText("Sorry, browser couldn't be opened.");
```
Comment (arnav-ag) [inline src/main/java/seedu/address/ui/ContactBox.java:46]: Does IOException only happen when browser cannot be opened? But I think it's also okay to just show this error as user might not care/need to know if any other error happens.

**Review by arnav-ag — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/ui/ContactBox.java (hunk 1 above)
Comment (Tex-Tang) [reply src/main/java/seedu/address/ui/ContactBox.java:49]: Any suggestion? we need to show at least a friendly message.

File: src/main/java/seedu/address/ui/ContactBox.java (hunk 2 above)
Comment (Tex-Tang) [reply src/main/java/seedu/address/ui/ContactBox.java:46]: Any suggestion? we need to show at least a friendly message.

**Review by arnav-ag — APPROVED** [review]
LGTM

**Pull request merged**